flang/lib/Semantics/resolve-names.cpp

```diff
@@ -8854,6 +8854,9 @@ void ResolveNamesVisitor::Post(const parser::AssignedGotoStmt &x) {
 }
 
 void ResolveNamesVisitor::Post(const parser::CompilerDirective &x) {
+  if (const auto *dir{
```

Comment:
`dir` is not used.  `if (std::holds_alternative<parser::CompilerDirective::VectorAlways>(x.u)) {` suffices.

Braces are missing.  Always use braces around `if`/`for`/`while`/&c. bodies in the modern C++ bits of f18.

Reply:
@DavidTruby can you make this change or comment here?